dlrover/python/master/diagnosis/diagnosis_manager.py

```diff
@@ -275,31 +275,48 @@ def start_metric_collect(self):
         store the data into global JobMetricContext
 
         """
-        logger.info(f"start {_dlrover_context.xpu_type} metric collector...")
-        if not os.getenv("DLROVER_METRIC_URL", ""):
+        if not self._job_args.metric_url:
+            _dlrover_context.metric_url = os.getenv("DLROVER_METRIC_URL", "")
+        else:
+            _dlrover_context.metric_url = self._job_args.metric_url
+
+        if not self._job_args.metric_token:
+            _dlrover_context.metric_token = os.getenv(
+                "DLROVER_METRIC_TOKEN", ""
+            )
+        else:
+            _dlrover_context.metric_token = self._job_args.metric_token
+
+        logger.info(f"start {self._job_args.xpu_type} metric collector...")
+        if not _dlrover_context.metric_url:
             logger.warning("no GPU metrics url defined, stop metric collector")
-            return
-        if not os.getenv("DLROVER_METRIC_TOKEN", ""):
+            return DiagnosisResult.DIAG_ERROR
+        if not _dlrover_context.metric_token:
```

Comment:
1. It is not necessarily required to have a token to obtain metrics for all cases.
2. Additionally, the message "no GPU metrics token defined, stop metric collector" seems a bit strange. Does it mean that only GPU metrics require a token? Other metrics, such as CPU metrics, don't require a token?